lib/nvme/nvme_ctrlr.c

```diff
@@ -849,6 +841,9 @@ nvme_ctrlr_process_init(struct spdk_nvme_ctrlr *ctrlr)
 			cc.bits.en = 0;
 			nvme_mmio_write_4(ctrlr, cc.raw, cc.raw);
 			nvme_ctrlr_set_state(ctrlr, NVME_CTRLR_STATE_DISABLE_WAIT_FOR_READY_0, ready_timeout_in_ms);
+
+			if (ctrlr->quirks & NVME_QUIRK_DELAY_BEFORE_CHK_RDY)
+				sleep(2);
```

Comment:
Thanks. Very good point. 
Adding a new state may not help, since at the very beginning of nvme_ctrlr_process_init(), it accesses registers, which shall be avoided.
The solution is adding a new field sleep_timeout_in_ms in spdk_nvme_ctrlr, which is checked in nvme_ctrlr_process_init(). Please see the new patch for details.